src/plugins/intel_cpu/src/nodes/mkldnn_rnn.cpp

```diff
@@ -725,21 +717,67 @@ void MKLDNNRNN::createDescriptor(const std::vector<MemoryDescPtr> &inputDesc,
     supportedPrimitiveDescriptors.emplace_back(config, ref_any);
 }
 
-void MKLDNNRNN::createPrimitive() {
+void MKLDNNRNN::prepareParams() {
+    auto dataMemPtr = getParentEdgesAtPort(0).front()->getMemoryPtr();
+    if (!dataMemPtr)
+        THROW_ERROR << "has uninitialized memory at port 0.";
+
+    const auto dataType = MKLDNNExtensionUtils::IEPrecisionToDataType(runtimePrecision);
+
+    const size_t B = dataMemPtr->GetShape().getStaticDims()[0];
+    const size_t SL = is_cell ? 1lu : dataMemPtr->GetShape().getStaticDims()[1];
+    const Shape shapeS_4D{L, D, B, SC};
+
+    inDataDescs[0] = DnnlBlockedMemoryDesc(Shape{SL, B, DC}, dataType, memory::format_tag::tnc);
+    outDataDescs[0] = DnnlBlockedMemoryDesc(Shape{SL, B, SC}, dataType, memory::format_tag::tnc);
+
+    inDataDescs[1] = DnnlBlockedMemoryDesc(shapeS_4D, dataType, memory::format_tag::ldnc);
+    outDataDescs[1] = DnnlBlockedMemoryDesc(shapeS_4D, dataType, memory::format_tag::ldnc);
+
+    if (haveCellState(cell_type)) {
+        inDataDescs[2] = DnnlBlockedMemoryDesc(shapeS_4D, memory::data_type::f32, memory::format_tag::ldnc);
+        outDataDescs[2] = DnnlBlockedMemoryDesc(shapeS_4D, memory::data_type::f32, memory::format_tag::ldnc);
+    }
+
+    bool wFormatWasChanged = false;
+    // WA To avoid different weights layer and iter formats in FP32 case.
+    if (runtimePrecision == Precision::FP32) {
+        if (SL != 1 || B < optimalBatchSize) {
+            if (wFormat != mkldnn::memory::format_tag::ldigo) {
+                wFormat = mkldnn::memory::format_tag::ldigo;
+                wFormatWasChanged = true;
+            }
+        } else if (wFormat != mkldnn::memory::format_tag::any) {
+            wFormat = mkldnn::memory::format_tag::any;
+            wFormatWasChanged = true;
+        }
+    }
+    if (wFormatWasChanged) {
+        auto weightsDims = MKLDNNExtensionUtils::convertToDnnlDims(VectorDims{ L, D, DC, G, SC });
+        wDescs[0] = mkldnn::memory::desc(weightsDims, dataType, wFormat);
+        auto statesDims = MKLDNNExtensionUtils::convertToDnnlDims(VectorDims{ L, D, SC, G, SC });
+        wDescs[1] = mkldnn::memory::desc(statesDims, dataType, wFormat);
+    }
+
+    fillDescs();
     if (cell_type == mkldnn::algorithm::vanilla_rnn) {
-        auto prim_desc = createPrimitiveDescriptor<vanilla_rnn_forward::primitive_desc, vanilla_rnn_forward::desc>();
-        prim.reset(new vanilla_rnn_forward(prim_desc));
+        std::shared_ptr<vanilla_rnn_forward::desc> desc = descs[0];
```

Comment:
shouldn't we iterate through all possible implementation like in other nodes?
`auto itpd = desc.createPrimitiveDescriptorIterator(engine, attr);`
`while (static_cast<bool>(itpd))  {`
    `// do something`
    `if (!itpd.next_impl())`
        `break;`
`}`

Reply:
Actually there is just one implementation type. No need to compare it with selected PD. Primitive and descs will be recreated anyway.